src/layout/node.ts

```diff
@@ -20,17 +22,23 @@ export default function(node: Node, source: Source): void {
         node = node.parentNode;
     }
 
-    let call = dom.has(node) ? "update" : "add";
+    let add = dom.has(node) === false;
+    let call = add ? "add" : "update";
     let parent = node.parentElement ? node.parentElement : null;
+    let insideFrame = node.ownerDocument !== document;
     switch (node.nodeType) {
-        case Node.DOCUMENT_NODE:
-            observe(node);
-            break;
         case Node.DOCUMENT_TYPE_NODE:
+            // Skip processing nodes within iframe if they were already removed from the DOM before we got to processing them.
+            // In those cases, we skip them and continue processing rest of the nodes.
+            let frame = insideFrame && node.parentNode ? dom.iframe(node.parentNode) : null;
```

Comment:
What happens with multiple nested iframes?

Reply:
It works and yes the changes we have handle it. I've tried on few pages where we have iframe inside iframe inside iframe and so far haven't found issues in my testing.